Enable ARM branch protection in gcc + restore OpenSSF hardening in glibc

dannf opened the pull request:
Fixes: https://github.com/chainguard-dev/internal-dev/issues/7940

dannf committed:
gcc: Enable ARM branch protection (BTI/PAC)

Signed-off-by: dann frazier <dann.frazier@canonical.com>

xnox commented on pipelines/test/compiler-hardening-check.yaml:
```diff
@@ -34,7 +34,6 @@ pipeline:
       ${{inputs.cc}} -v -o hello-default hello.c
       out=$(./hello-default 1 || true)
-      [ "$out" = "hello-c" ]
 
```
this checks that compiled thing is executable.

dannf committed:
glibc: Re-enable OpenSSF hardening

glibc was failing to build with openssf-compiler-options
on ARM because those options enable branch protection (BTI/PAC).
This feature can only work if all objects are built with
BTI/PAC, and that was not the case with the GCC code that
gets linked into glibc. GCC has been fixed, so let's
restore hardening.

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

dannf committed:
Disable the branchprotection test on ARM until glibc is rebuilt with it

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>

xnox commented:
AMAZING
